action.yml

```diff
@@ -1,57 +1,51 @@
-name: 'Allure Report with history per branch'
-description: 'Generate Allure Report with history per branch'
-author: 'mgrybyk'
-branding:
-  icon: 'monitor'
-  color: 'gray-dark'
-inputs:
-  report_dir:
-    description: 'Input allure results dir, default "allure-results"'
-    required: true
-    default: 'allure-results'
-  report_id:
-    description: 'Output allure report dir, default "allure-report"'
-    required: true
-    default: 'allure-report'
-  gh_pages:
-    description: 'Folder with gh-pages branch'
-    required: true
-    default: 'gh-pages'
-  list_dirs:
-    description: 'Write index.html to the Github Action folders. Might cause concurrency issues!'
-    required: false
-    default: false
-  list_dirs_branch:
-    description: 'Write index.html to the report branch folder. Might cause concurrency issues!'
-    required: false
-    default: true
-  branch_cleanup_enabled:
-    description: 'Cleanup reports from deleted branches. Might cause concurrency issues!'
-    required: false
-    default: false
-  max_reports:
-    description: 'Max reports to keep per branch/report. Set to 0 to disable cleanup.'
-    required: false
-    default: 100
-outputs:
-  report_url:
-    description: 'Published Allure Report url'
-  report_history_url:
-    description: 'Published Allure History list url'
-  test_result:
-    description: 'test result: PASS | FAIL | UNKNOWN'
-  test_result_icon:
-    description: 'test result icon: ✅ | ❌ | ❔'
-  test_result_passed:
-    description: 'Number of passed tests'
-  test_result_failed:
-    description: 'Number of failed tests'
-  test_result_total:
-    description: 'Number of tests in total'
-  run_unique_id:
-    description: 'Name of the folder containing the generated Allure Report'
-  report_path:
-    description: 'Path to the folder containing the generated Allure Report relative to the repo root'
-runs:
-  using: 'docker'
-  image: 'Dockerfile'
+name: 'Allure Report with history per branch'
+description: 'Generate Allure Report with history per branch'
+author: 'mgrybyk'
+branding:
+  icon: 'monitor'
+  color: 'gray-dark'
+inputs:
+  prev_git_hash:

```

Comment:
Thinking about this more, it might make sense to use a "list" of previous Git hashes instead just the single previous hash. The reason being is that not all commits might have test reports/history associated with them, so we'd have to fall back to older commits if necessary. One option is to wait and see if this change will be necessary and do it under another card if so.